presto-postgresql/src/test/java/com/facebook/presto/unittests/TestPostgresqlMetadata.java

```diff
@@ -0,0 +1,108 @@
+import com.facebook.presto.connector.unittest.BaseMetadataTest;
+import com.facebook.presto.connector.unittest.MetadataSchemaTest;
+import com.facebook.presto.connector.unittest.MetadataTableTest;
+import com.facebook.presto.plugin.postgresql.PostgreSqlPlugin;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorFactory;
+import com.facebook.presto.testing.TestingConnectorContext;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.testing.postgresql.TestingPostgreSqlServer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.connector.meta.ConnectorFeature.CREATE_TABLE_AS;
+import static com.facebook.presto.connector.meta.ConnectorFeature.DROP_TABLE;
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+@SupportedFeatures({
+        CREATE_TABLE_AS,
+        DROP_TABLE})
+public class TestPostgresqlMetadata
+        implements BaseMetadataTest, MetadataTableTest, MetadataSchemaTest
+{
+    private TestingPostgreSqlServer server;
+    private Connector connector;
+
+    @BeforeAll
+    public void beforeAll()
+            throws Exception
+    {
+        this.server = new TestingPostgreSqlServer("testuser", "unittests");
+        this.connector = createPostgresqlConnector(server);
+    }
+
+    @AfterAll
+    public void cleanUp()
+            throws Exception
+    {
+        connector.shutdown();
+        server.close();
+    }
+
+    @Override
+    public Connector getConnector()
+    {
+        return connector;
+    }
+
+    private static Connector createPostgresqlConnector(TestingPostgreSqlServer server)
+            throws Exception
+    {
+        PostgreSqlPlugin plugin = new PostgreSqlPlugin();
+        Iterable<ConnectorFactory> connectorFactories = plugin.getConnectorFactories();
+
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("connection-url", server.getJdbcUrl())
+                .put("allow-drop-table", "true")
+                .build();
+
+        ConnectorFactory factory = getOnlyElement(connectorFactories);
+        return factory.create("postgresql", properties, new TestingConnectorContext());
+    }
+
+    @Override
+    public Map<String, Object> getTableProperties()
+    {
+        return ImmutableMap.of();
+    }
+
+    @Override
+    public SchemaTableName schemaTableName(String tableName)
+    {
+        return new SchemaTableName("public", tableName);
+    }
+
+    @Override
+    public List<String> systemSchemas()
+    {
+        return ImmutableList.of("pg_catalog", "public");
+    }
+
+    @Override
+    public List<AutoCloseable> withSchemas(ConnectorSession session, List<String> schemaNames)
```

Comment:
Commented.

In case it isn't clear from the comments:

> Shouldn't only schema related tests use the withSchemas method.

No. Anything that requires a table requires a schema. For e.g. Hive, you need to poof one into existence and then destroy it. For e.g. PostgreSQL, you already have it.